clients/java/src/main/java/io/unitycatalog/client/ApiClientBuilder.java

```diff
@@ -113,30 +127,47 @@ public ApiClientBuilder retryPolicy(RetryPolicy retryPolicy) {
     return this;
   }
 
+  public ApiClientBuilder addRequestInterceptor(Consumer<HttpRequest.Builder> interceptor) {
```

Comment:
Organizing the `interceptor` as a queue with the insertion order, and then intercepting one by one.  This will help a lot for multiple interceptions.